### ISSUE TITLE
batch norm broken

### ISSUE THREAD

**Issue by aluo-x** [aluo-x/tensorflow_windows#6]
Seems like batch norm is still broken in the Tensorflow 1.4.0 compiled for GPU

**Comment by aluo-x** [comment]
```
2017-12-04 01:12:16.941698: I C:\optimae\tensorflow-1.4.0\tensorflow\core\common_runtime\gpu\gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: GeForce GTX 1050, pci bus id: 0000:01:00.0, compute capability: 6.1)
2017-12-04 01:12:20.142008: I C:\optimae\tensorflow-1.4.0\tensorflow\core\common_runtime\gpu\gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: GeForce GTX 1050, pci bus id: 0000:01:00.0, compute capability: 6.1)
2017-12-04 01:12:42.643955: W C:\optimae\tensorflow-1.4.0\tensorflow\core\framework\op_kernel.cc:1192] Invalid argument: Incompatible shapes: [4] vs. [0]
	 [[Node: training/Adam/gradients/batch_normalization_1/moments/mean_grad/floordiv = FloorDiv[T=DT_INT32, _class=["loc:@batch_normalization_1/moments/mean"], _device="/job:localhost/replica:0/task:0/device:GPU:0"](training/Adam/gradients/batch_normalization_1/moments/mean_grad/Shape, training/Adam/gradients/batch_normalization_1/moments/mean_grad/Maximum)]]
2017-12-04 01:12:42.644503: W C:\optimae\tensorflow-1.4.0\tensorflow\core\framework\op_kernel.cc:1192] Invalid argument: Incompatible shapes: [4] vs. [0]
	 [[Node: training/Adam/gradients/batch_normalization_1/moments/mean_grad/floordiv = FloorDiv[T=DT_INT32, _class=["loc:@batch_normalization_1/moments/mean"], _device="/job:localhost/replica:0/task:0/device:GPU:0"](training/Adam/gradients/batch_normalization_1/moments/mean_grad/Shape, training/Adam/gradients/batch_normalization_1/moments/mean_grad/Maximum)]]
2017-12-04 01:12:42.645249: W C:\optimae\tensorflow-1.4.0\tensorflow\core\framework\op_kernel.cc:1192] Invalid argument: Incompatible shapes: [4] vs. [0]
```

**Comment by aluo-x** [comment]
```
InvalidArgumentError (see above for traceback): Incompatible shapes: [4] vs. [0]
	 [[Node: training/Adam/gradients/batch_normalization_1/moments/mean_grad/floordiv = FloorDiv[T=DT_INT32, _class=["loc:@batch_normalization_1/moments/mean"], _device="/job:localhost/replica:0/task:0/device:GPU:0"](training/Adam/gradients/batch_normalization_1/moments/mean_grad/Shape, training/Adam/gradients/batch_normalization_1/moments/mean_grad/Maximum)]]
	 [[Node: loss/output_large_loss/Mean_2/_791 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device_incarnation=1, tensor_name="edge_7673_loss/output_large_loss/Mean_2", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]
```
